src/main/java/it/smartcommunitylab/aac/webauthn/service/WebAuthnYubicoCredentialsRepository.java

```diff
@@ -0,0 +1,121 @@
+package it.smartcommunitylab.aac.webauthn.service;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.yubico.webauthn.CredentialRepository;
+import com.yubico.webauthn.RegisteredCredential;
+import com.yubico.webauthn.data.AuthenticatorTransport;
+import com.yubico.webauthn.data.ByteArray;
+import com.yubico.webauthn.data.PublicKeyCredentialDescriptor;
+import com.yubico.webauthn.data.PublicKeyCredentialType;
+
+import org.springframework.util.StringUtils;
+
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnCredential;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnUserAccount;
+
+/**
+ * For this class, everytime we should use the 'username', we
+ * will instead use a string in the following format:
+ * 'realmname{@link #separator}username'. This
+ * is due to the fact that yubico's library assumes that two distinct users can
+ * not have the same username (https://git.io/JD5Vr).
+ */
+public class WebAuthnYubicoCredentialsRepository implements CredentialRepository {
+
+    private final String providerId;
+    private WebAuthnUserAccountService webAuthnUserAccountService;
```

Comment:
field is final

Reply:
Addressed in 7083e6860230f9dfaab6f71b5ba7338caab4e077